### ISSUE TITLE
Select: throws an error with auto-filter-focus

### ISSUE THREAD

**Issue by Tamas-hi** [primefaces/primevue#6539]
### Describe the bug

From v 4.10, the Select component throws an error when you click out of the dropdown, when auto-filter-focus is enabled.

<PrimeVueSelect
	v-model="customer"
	checkmark
	show-clear
	filter
	**auto-filter-focus** // this throws an error
	scroll-height="20rem"
	option-label="company_code"
	:options="customerOptions"
	:loading="customerOptionsLoading"
	:disabled="customerOptionsLoading"
	class="w-full"
	:invalid="Boolean(errors.customer)"
	v-bind="customerAttrs"
>
</PrimeVueSelect>

![image](https://github.com/user-attachments/assets/050f024c-1156-4eb4-92fe-1d7abd542f32)

### Reproducer

-

### PrimeVue version

4.10

### Vue version

4.x

### Language

TypeScript

### Build / Runtime

Vite

**Comment by 1Map** [comment]
The following line in your code throws an error as filterInput is null:

`focus(this.$refs.filterInput.$el);`

[primevue/packages/primevue/src/select/Select.vue](https://github.com/primefaces/primevue/blob/22d6f8a03724f30b6695a0800412f77a3a8ed738/packages/primevue/src/select/Select.vue#L710)

Line 710 in [22d6f8a](https://github.com/primefaces/primevue/commit/22d6f8a03724f30b6695a0800412f77a3a8ed738)

`focus(this.$refs.filterInput.$el); `
 
I do not think it should try and focus on the filter during onOverlayLeave as the filterInput is not always there (null)

**Comment by 1Map** [comment]
The following should be completely removed from `onOverlayLeave` as there is no need to focus on the filterInput Element during Leave.

```
            if (this.autoFilterFocus && this.filter) {
                this.$nextTick(() => {
                    focus(this.$refs.filterInput.$el);
                });
            }
```

It is only needed in `onOverlayEnter`

**Comment by wijzijnweb** [comment]
Thought the same @1Map. Created a pull-request https://github.com/primefaces/primevue/pull/6546 but they considered fixing it the focus feature.
But in my opinion as well there seems no need to focus onLeave since the input isn't shown anyways.
Better to focus the document instead, but I don't think it matters.

We'll see how it will be fixed.

**Comment by 1Map** [comment]
@wijzijnweb Thanks, well still not fixed in the latest version

**Comment by Tamas-hi** [comment]
Still in an issue in 4.2.2, can anyone else confirm it?
![image](https://github.com/user-attachments/assets/fb1b6074-ccb9-4e97-af0e-2e92a36e5995)

**Comment by lakkvak** [comment]
> Still in an issue in 4.2.2, can anyone else confirm it? ![image](https://private-user-images.githubusercontent.com/36476318/386194451-fb1b6074-ccb9-4e97-af0e-2e92a36e5995.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NzYwNTUsIm5iZiI6MTczMTY3NTc1NSwicGF0aCI6Ii8zNjQ3NjMxOC8zODYxOTQ0NTEtZmIxYjYwNzQtY2NiOS00ZTk3LWFmMGUtMmU5MmEzNmU1OTk1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE1VDEzMDIzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0MGU5YWM4ZTA1OTQ3MGRjM2M3MmQxMmIxOWEzOTQ3YzQ5ODI0NmIxM2ZhNWI0MzRiZGNmM2MxMDRmYTI3ODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1J-blXRWIKBa-WkJXWblyxQRbHy9ptOw1Ll28fliB-k)

yes, same

**Comment by wijzijnweb** [comment]
Can confirm, it's still an issue onLeave.

**Comment by danve** [comment]
Still there on 4.2.5 version

**Comment by 1Map** [comment]
I can confirm.  I already made suggestion almost 2 months ago:

https://github.com/primefaces/primevue/issues/6539#issuecomment-2415786460

But it seems it is ignored.

**Comment by andrescellini** [comment]
keeps happening 2025.

**Comment by valoup917** [comment]
Yeah still having the issue

**Comment by nahojj** [comment]
+1

**Comment by haizk** [comment]
fixed 4.3.0

#7186
